app/models/discussion_reader.rb

```diff
@@ -18,17 +18,27 @@ def self.for(user: , discussion: )
     end
   end
 
+  def self.for_comment(comment:)
```

Comment:
I'm iterating this in the next pass to accept a vanilla model which has an author and a discussion

```
def self.for_model(model)
  self.for(user: model.author, discussion: model.discussion)
end
```